common/sgx/report.c

```diff
@@ -92,7 +93,7 @@ oe_result_t oe_parse_report(
         OE_RAISE(OE_INVALID_PARAMETER);
 
     if (header->report_size + sizeof(oe_report_header_t) != report_size)
-        OE_RAISE(OE_FAILURE);
+        OE_RAISE(OE_INCORRECT_REPORT_SIZE);
```

Comment:
This seems extremely specific to warrant its own error code. I would suggest using `OE_QUOTE_VERIFICATION_ERROR` and performing an OE_RAISE_MSG instead to pinpoint the specific case. Similarly for the ISVPRODID check above and others.

Reply:
The problem with OE_RAISE_MSG is that the message will now show up in production (since it is not running in debug mode).

Reply:
Error codes are generally a developer facing feature, so I'm not sure why the production distinction is important here? Under what conditions in production are you expecting to be able to retrieve the OE_RESULT value and interpret it, and who is your audience for this specific error code if not maintainers of OE? (i.e. there's nothing an app developer can do with this error state, is there?)

Reply:
I think it will help faster diagnostics when debugging issues.